packages/dataviews/src/types.ts

```diff
@@ -205,6 +205,16 @@ interface ActionBase {
 }
 
 export interface ActionModal extends ActionBase {
+	/**
+	 * The callback to execute when the action has finished.
+	 */
+	onActionPerformed: ( ( items: Item[] ) => void ) | undefined;
+
+	/**
+	 * The callback to execute when the action is triggered.
+	 */
+	onActionStart: ( ( items: Item[] ) => void ) | undefined;
```

Comment:
Are these properties specific to modal actions or should they be moved to ActionBase

Reply:
I only see them in use by RenderModal https://github.com/WordPress/gutenberg/blob/trunk/packages/dataviews/src/item-actions.tsx#L119

Reply:
Makes sense. Given the names I think they can be supported in any type of action but since it follows our current implementation, it's fine to keep like that for now.